src/pages/index.js

```diff
@@ -90,19 +187,88 @@ function openEditCardModal() {
   addCardModal.open();
 }
 
+function openProfileImageModal() {
+  editAvatarModal.open();
+}
+
 /*------------------------------------------------------------------*/
-/*                          Event Objects                            */
+/*                          EVENT HANDLERS                            */
 /*------------------------------------------------------------------*/
 
 function handleProfileEditSubmit({ title, description }) {
-  userInfo.setUserInfo({ title, description });
-  profileEditModal.close();
+  profileEditModal.setLoading(true);
+  api.updateProfileInfo({ title, description }).then((res) => {
+    profileUserInfo.setUserInfo({ title, description });
+    profileEditModal.setLoading(false);
+    profileEditModal.close();
+  })
+  .catch((err) => {
+      console.log(err);
+  });
+}
+
+function likeCard(card) {
+  api
+    .likeCard(card.getId())
+    .then(() => {
+      card.handleLikeIcon();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
 }
 
+function removeLikeCard(card) {
+  api
+    .removeLikeCard(card.getId())
+    .then(() => {
+      card.handleLikeIcon();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+}
+
+function handleDeleteClick(card) {
+  // open the delete card popup
+  deleteCardModal.open();
+  deleteCardModal.setSubmitCallback(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.handleDeleteCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+    deleteCardModal.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well